feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -315,6 +317,94 @@ class BillingWrapperTest {
         )
     }
 
+    @Test
+    fun `obfuscatedAccountId is set for non-transfer purchases`() {
+        setup()
+        val appUserID = "jerry"
+
+        val skuDetails = stubSkuDetails(productId = "product_a")
+        val activity: Activity = mockk()
+
+        mockkStatic(BillingFlowParams::class)
+        val mockBuilder = mockk<BillingFlowParams.Builder>(relaxed = true)
+        every {
+            BillingFlowParams.newBuilder()
+        } returns mockBuilder
+
+        every {
+            mockBuilder.setSkuDetails(any())
+        } returns mockBuilder
+
+        val params = mockk<BillingFlowParams>(relaxed = true)
+        every {
+            mockBuilder.build()
+        } returns params
+
+        every {
+            mockClient.launchBillingFlow(any(), params)
+        } returns BillingClient.BillingResponseCode.OK.buildResult()
+
+        billingClientStateListener!!.onBillingSetupFinished(BillingClient.BillingResponseCode.OK.buildResult())
+
+        wrapper.makePurchaseAsync(
+            activity,
+            appUserID,
+            skuDetails.toProductDetails(),
+            null,
+            null
+        )
+
+        verify {
+            mockBuilder.setObfuscatedAccountId(appUserID.sha256())
+        }
+
+        clearStaticMockk(BillingFlowParams::class)
+    }
+
+    @Test
+    fun `obfuscatedAccountId is not set for transfer purchases`() {
+        setup()
+        val appUserID = "jerry"
+
+        val skuDetails = stubSkuDetails(productId = "product_a")
+        val activity: Activity = mockk()
+
+        mockkStatic(BillingFlowParams::class)
+        val mockBuilder = mockk<BillingFlowParams.Builder>(relaxed = true)
+        every {
+            BillingFlowParams.newBuilder()
+        } returns mockBuilder
+
+        every {
+            mockBuilder.setSkuDetails(any())
+        } returns mockBuilder
+
+        val params = mockk<BillingFlowParams>(relaxed = true)
+        every {
+            mockBuilder.build()
+        } returns params
+
+        every {
+            mockClient.launchBillingFlow(any(), params)
```

Comment:
this setup is exactly the same for both tests, right? maybe we could extract it into a method, since it's a bit complex?